pdfminer/layout.py

```diff
@@ -422,7 +422,8 @@ def find_neighbors(self, plane, ratio):
                 if (isinstance(obj, LTTextLineHorizontal) and
                     abs(obj.height-self.height) < d and
                     (abs(obj.x0-self.x0) < d or
-                     abs(obj.x1-self.x1) < d))]
+                     abs(obj.x1-self.x1) < d or
+                     abs((obj.x0+obj.x1)/2-(self.x0+self.x1)/2) < d))]
```

Comment:
It always takes me a few minutes to grasp the meaning of all these comparisons. Could you introduce well-named (private) methods on `LTTextLineHorizontal` for these comparisons?